app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,25 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      report_id = ReportTemplate.find_by(name: "Subscription - Entitlement Report").id
+      days_from_now = TemplateInput.find_by_name("Days from Now").id.to_s
+      mail_query = UserMailNotification.first.mail_query.to_s
```

Comment:
The `mail_query` is available to us from the `SubscriptionMailer` so this method could be called from the mailer class with a new argument providing the mail_query

Reply:
Yes! That's a great idea. thanks.